sdk/signalr/Microsoft.Azure.WebJobs.Extensions.SignalRService/src/Config/IConfigurationExtensions.cs

```diff
@@ -65,5 +68,45 @@ public static bool TryGetNamedEndpointFromIdentity(this IConfigurationSection se
             endpoint = null;
             return false;
         }
+
+        public static bool TryGetJsonObjectSerializer(this IConfiguration configuration, out ObjectSerializer serializer)
+        {
+            var hubProtocolConfig = configuration[Constants.AzureSignalRHubProtocol];
+            //indicates Newtonsoft, camcelCase
+            if (configuration.GetValue(Constants.AzureSignalRNewtonsoftCamelCase, false))
+            {
+                serializer = new NewtonsoftJsonObjectSerializer(new JsonSerializerSettings()
+                {
+                    ContractResolver = new CamelCasePropertyNamesContractResolver()
+                });
+                return true;
+            }
+
+            if (configuration.OnDotnetInProcessRuntime() is not true)
+            {
+                serializer = new NewtonsoftJsonObjectSerializer();
+                return true;
+            }
+
+            if (hubProtocolConfig is not null)
+            {
+                serializer = Enum.Parse(typeof(HubProtocol), hubProtocolConfig, true) switch
+                {
+                    HubProtocol.NewtonsoftJson => new NewtonsoftJsonObjectSerializer(),
+                    HubProtocol.SystemTextJson => new JsonObjectSerializer(),
+                    _ => throw new InvalidOperationException(),
```

Comment:
default `_` will never be hit here and invalid value will throw exception during Enum.Parse, is this expected? Or shall we call `TryParse` and throw exception for `false` case.

Reply:
Yes, it is expected to throw exception if users set an invalid value. I think this make the code more simple.

Reply:
Throw `NotSupportedException`.